api/src/main/java/org/xwiki/apps/googleapps/GoogleAppsAuthService.java

```diff
@@ -18,26 +16,20 @@
  * License along with this software; if not, write to the Free
  * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
  * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
--->
+ */
+package org.xwiki.apps.googleapps;
+
+import org.xwiki.component.annotation.Role;
+import com.xpn.xwiki.user.api.XWikiAuthService;
 
-<xwikidoc>
-  <web>GoogleApps</web>
-  <name>AuthService</name>
-  <language/>
-  <defaultLanguage/>
-  <translation>0</translation>
-  <parent>GoogleApps.WebHome</parent>
-  <creator>xwiki:XWiki.Admin</creator>
-  <author>xwiki:XWiki.Admin</author>
-  <customClass/>
-  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
-  <version>1.1</version>
-  <title/>
-  <defaultTemplate/>
-  <validationScript/>
-  <comment/>
-  <minorEdit>false</minorEdit>
-  <syntaxId>xwiki/2.1</syntaxId>
-  <hidden>true</hidden>
-  <content/>
-</xwikidoc>
+/**
+ * A badge interface to denote the role of the component that will replace the
+ * default authentication-service.
+ *
+ * @since 3.0
+ * @version $Id$
+ */
+@Role
+public interface GoogleAppsAuthService extends XWikiAuthService
+{
+}
```

Comment:
Hello @acotiuga and @mflorea, 

yes, indeed, this is used at line 163 of GoogleAppsManagerImpl where the component's "role" (this interface) is invoked and has as sole implementation the one in the internal package.

There is no second allocation of the XWikiAuthService except through this place since there is no other configuration indicating it's use, except for the fact that the application is loaded. 

While I see the somewhat "too simple" aspect of this badge interface, it seems to be a strict application of the [Component Module](https://extensions.xwiki.org/xwiki/bin/view/Extension/Component%20Module). Have I misinterpreted something?

Reply:
@polx my bad, I was thinking the ``GoogleAppsAuthService`` is still registered through the XWiki configuration files, but it's not the case anymore.

Now, the issue with the 2 ``GoogleAppsManagerImpl`` instances you mentioned in some other place it's due to the fact that ``GoogleAppsManagerImpl`` implements multiple roles, and ATM the component manager creates a different instance for each role. So there's an ``GoogleAppsManagerImpl`` instance for the ``GoogleAppsManager`` role and another one for the ``EventListener`` role. ``Initializable`` and ``Disposable`` are not roles.

Note also that ``EventListener``s are initialized very early, before the XWiki database is ready, which explains why you can have ``null`` XWiki context in the initialization method of ``GoogleAppsManagerImpl``. The general recommendation in this case is to move the listener code in a separate component, e.g. ``GoogleAppsListener``.

Reply:
Done in 093bd59f4878dffc52fa3fefdf8bfbf14ff9126c: There is now a separate GoogleAppsEventListener which is registered at initialialization of GoogleApps. After a short chat, it seems that EventListener has a reason to be a role and thus I have made this component only bound to the lifecycle of the GoogleAppsManager lifecycle.

Similarly, the GoogleAppsAuthService (that badge interface commented upon here) has an implementation that is registered at initialization of the GoogleAppsManager. Because its instatiation is per-lookup, one can take advantage of components (e.g. get a logger) but is not bent to an independent (possibly half-complete) initialization. The duplicate or bad-initialized components have not been encountered anymore in my testings.

Thanks.
paul